src-plugins/mscExportVideo/mscExportVideo.cpp

```diff
@@ -0,0 +1,283 @@
+        writerJPEG->SetFileName(name.toStdString().c_str());
+        writerJPEG->SetInputConnection(vtkImage->GetProducerPort());
+        writerJPEG->Write();
+
+        cpt++;
+    }
+
+    return medAbstractProcess::SUCCESS;
+}
+
+int ExportVideo::exportAsVideo()
+{
+    vtkSmartPointer<vtkImageCanvasSource2D> source = vtkSmartPointer<vtkImageCanvasSource2D>::New();
+    source->SetScalarTypeToUnsignedChar();
+    source->SetNumberOfScalarComponents(3);
+    source->SetExtent(0, d->width-1, 0, d->height-1, 0, 0);
+
+    if (d->format == OGGVORBIS)
+    {
+        vtkSmartPointer<vtkOggTheoraWriter> writerVideo = vtkSmartPointer<vtkOggTheoraWriter>::New();
+        writerVideo->SetInputConnection(source->GetOutputPort());
+        writerVideo->SetFileName(d->filename.toStdString().c_str());
+        writerVideo->SetRate(d->frameRate);
+        writerVideo->SetSubsampling(d->subsampling); // OGG/OGV parameter
+        writerVideo->SetQuality(d->quality);
+        writerVideo->Start();
+
+        foreach (QImage qimage, d->imagesArray)
+        {
+            // Qt to VTK images (vtkImageData)
+            vtkSmartPointer<vtkQImageToImageSource> qimageToImageSource = vtkSmartPointer<vtkQImageToImageSource>::New();
+            qimageToImageSource->SetQImage(&qimage);
+            qimageToImageSource->Update();
+            vtkImageData* vtkImage = qimageToImageSource->GetOutput();
+
+            source->DrawImage(0, 0, vtkImage);
+            source->Update();
+
+            writerVideo->Write();
+        }
+
+        writerVideo->End();
+
+        return medAbstractProcess::SUCCESS;
+    }
+
+    return medAbstractProcess::FAILURE;
+}
+
+int ExportVideo::displayFileDialog()
+{
+    QFileDialog * exportDialog = new QFileDialog(0, tr("Save movie as"));
+    exportDialog->setOption(QFileDialog::DontUseNativeDialog);
+    exportDialog->setAcceptMode(QFileDialog::AcceptSave);
+    exportDialog->selectFile("video.ogv");
+
+    QLayout* layout = exportDialog->layout();
+    QGridLayout* gridbox = qobject_cast<QGridLayout*>(layout);
+
+    // Hide the filter list
+    QWidget * filtersLabel = gridbox->itemAtPosition(gridbox->rowCount()-1, 0)->widget();
+    QWidget * filtersList = gridbox->itemAtPosition(gridbox->rowCount()-1, 1)->widget();
+    filtersLabel->hide(); filtersList->hide();
+
+    // Add user widgets: Video Format
+    QComboBox* format = new QComboBox(exportDialog);
+    // Free, opensource, no patent.
+    // "Before 2007, the .ogg filename extension was used for all files whose content used the Ogg container
+    // format. Since 2007, the Xiph.Org Foundation recommends that .ogg only be used for Ogg Vorbis audio files.
+    // The Xiph.Org Foundation decided to create a new set of file extensions and media types to describe different
+    // types of content such as .oga for audio only files, .ogv for video with or without sound (including Theora),
+    // and .ogx for multiplexed Ogg."
+    format->addItem("Ogg Vorbis (.ogv)", 0);
```

Comment:
small note: you can read ogv files with a lot of soft, including Firefox and other browser.